app/controllers/dashboards_controller.rb

```diff
@@ -22,9 +31,29 @@ def export
     redirect_to dashboards_path, alert: e.message
   end
 
+  def push
+    version = Version.preview_version
+
+    if version.blank?
+      flash.alert = 'No preview version available'
+    else
+      pv = Version.create(number: version.number, production: true, user: current_user)
+
+      if pv.persisted?
+        flash.notice = 'Production data updated'
+      else
+
+        flash.alert = "Production data not updated: #{version.inspect} #{pv.errors.inspect}"
+      end
+    end
+
+    redirect_to dashboards_path
```

Comment:
do you really want to redirect to dashboards_path if there is an error? I would think you would want to render the same page with the specified params in case they want try again.